scripts/generator/graphql_generator/csharp/SDK/iOS/iOSWebCheckout.cs.erb

```diff
@@ -36,12 +51,42 @@ namespace <%= namespace %>.SDK.iOS {
             if (message.Content == "loadedThankYou") {
                 // Do a round trip to the server to see if the purchase is solid.
                 // Invoke callbacks on cart to pass to Unity
-                OnSuccess();
+                DidCheckoutCompleteSuccessfully(CurrentCheckout.id(), (didComplete) => {
+                    if (didComplete) {
+                        OnSuccess();
+                    } else {
+                        var error = new ShopifyError(
+                            ShopifyError.ErrorType.CheckoutError, 
+                            "User navigated to end of checkout without successfully checking out."
+                        );
+
+                        OnFailure(error);
+                    }
+                });
             } else if (message.Content == "cancelled") {
                 OnCancelled();
                 message.Respond("");
             }
         }
+
+        private delegate void CheckoutCheckCallback(bool didComplete);
+        private void DidCheckoutCompleteSuccessfully(string checkoutId, CheckoutCheckCallback callback) {
```

Comment:
Does this sound a bit like a delegate callback? Would `VerifyCheckoutCompleteSuccessfully` be a better name?